nntrainer/layers/layer_impl.h

```diff
@@ -52,7 +53,7 @@ class LayerImpl : public Layer {
   /**
    * @brief     finalize the layer
    * @throw     nntrainer::not_supported if try to initialize twice
-   * @copydoc   Layer::fianlize(InitLayerContext &context)
+   * @copydoc   Layer::finalize(InitLayerContext &context)
```

Comment:
I wonder that LayerImpl is the abstract class for all layers?? (Line 37) You mean the abstraction level of this LayerImpl is higher than Layer, I guess. It looks like developer can not make their own layer with only including this LayerImpl if they don't know about the layer_devel.h

Reply:
I am thinking that all the layers which have weights/bias will extend LayerImpl, while others will extend LayerDevel.
LayerImpl for now only provides support for Weight/bias-related properties, so layers like activation/flatten/etc can extend LayerDevel directly IMO. What do you advise?

Reply:
> It looks like developer can not make their own layer with only including this LayerImpl if they don't know about the layer_devel.h

Yes, that is absolutely right. The developer will have to look at LayerDevel. LayerImpl is also an abstract class.